StRoot/StFwdUtils/StFwdAnalysisMaker.h

```diff
@@ -27,6 +18,51 @@ class StFwdAnalysisMaker : public StMaker {
     void Clear(const Option_t *opts = "");
     void ProcessFwdTracks();
     void ProcessFwdMuTracks();
+
+    // StEvent analyzed by default
+    // call this to analyze the MuDst instead
+    void setMuDstInput() { mAnalyzeMuDst = true; }
+
+  protected:
+
+    /**
+     * @brief Map of <name (TString), histogram>
+     * 
+     */
+    std::map<TString, TH1*> mHists;
+
+    /**
+     * @brief Add a histogram to the map
+     * 
+     * Convenience method to avoid duplicate typing name / possible mismatch
+     * @param h1 TH1* histogram object
+     * @return TH1* return the same object for ease of use
+     */
+    TH1 * addHist( TH1 * h1){
+      mHists[h1->GetName()] = h1;
+      return h1;
```

Comment:
Return the same pointer that was passed to it? Seems a bit odd, but generally harmless I suppose.

Reply:
Ah....I see that you're using `new TH1/2F()` in the argument, so this let's you get that pointer if you need it, though I see you aren't currently using the returned pointer. Another approach would have been to define `AddHist()` to simply take either the arguments of a TH1F or TH2F constructor and call `new` inside `AddHist()`, since instantiating a new hist is what you always do, but your approach should work fine too.

Reply:
Removed this and instead use the `AddHist(...)` from StMaker
see: [030a2f5](https://github.com/star-bnl/star-sw/pull/636/commits/030a2f51a0cb1aa2368fac8ec6070243646aeda0)

Reply:
Even better!